internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -343,7 +345,7 @@ func addDefinitionToExistingStatusVariable(variable clusterv1.ClusterClassVariab
 	// If definitions already conflict, no need to check.
 	if !combinedVariable.DefinitionsConflict {
 		currentDefinition := combinedVariable.Definitions[0]
-		if !(currentDefinition.Required == newVariableDefinition.Required && reflect.DeepEqual(currentDefinition.Schema, newVariableDefinition.Schema)) {
+		if !(currentDefinition.Required == newVariableDefinition.Required && reflect.DeepEqual(currentDefinition.Schema, newVariableDefinition.Schema) && reflect.DeepEqual(currentDefinition.Metadata, newVariableDefinition.Metadata)) {
```

Comment:
note: I think this is ok for now, but in the future we can think about making this more tolerant to definitions with different metadata.
E.g. we merge metadata if there are no conflicting values

Reply:
That's a great idea.

Reply:
We intentionally made a different choice with the metadata we already have within the scheme (description, example, ...)

Reply:
The idea there was that we shouldn't merge something that might not be the same. Merging it could be risky if the semantic differs.